sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -414,6 +418,84 @@ class UDFRegistration private[sql] (functionRegistry: FunctionRegistry) extends
   //////////////////////////////////////////////////////////////////////////////////////////////
 
   /**
+   * Register a Java UDF class using reflection, for use from pyspark
+   *
+   * @param name   udf name
+   * @param className   fully qualified class name of udf
+   * @param returnDataType  return type of udf. If it is null, spark would try to infer
+   *                        via reflection.
+   */
+  def registerJava(name: String, className: String, returnDataType: DataType): Unit = {
```

Comment:
Is it possible to make this non-public?  I believe we do this in other cases for code only called from python.